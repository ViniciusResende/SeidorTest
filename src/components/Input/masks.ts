export function cpf(e: React.FormEvent<HTMLInputElement>){
  e.currentTarget.maxLength = 14;
  let value = e.currentTarget.value;
  value = value.replace(/\D/g, "");
  // value = value.replace(/(\d)(\d{2})$/, "$1-$2");
  // value = value.replace(/(?=(\d{3})+(\D))\B/g, ".")
  value = value.replace(/^(\d{3})(\d{3})(\d{3})(\d)/, "$1.$2.$3-$4")
  e.currentTarget.value = value;

  return e;
}

export function currency(e: React.FormEvent<HTMLInputElement>){
  let value = e.currentTarget.value;
  value = value.replace(/\D/g, "");
  value = value.replace(/(\d)(\d{2})$/, "$1,$2");
  value = value.replace(/(?=(\d{3})+(\D))\B/g, ".")
  e.currentTarget.value = value;

  return e;
}