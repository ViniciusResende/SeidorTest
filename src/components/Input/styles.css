.input-group {
  display: flex;
}
.prefix-span {
  display: flex;
  align-items: center;

  padding: 6px 12px;
  font-size: 14px;
  color: var(--gray-800);
  background-color: var(--ice);
  box-shadow: 0px 2px 2px rgba(187, 204, 221, 0.4);
}

input {
  flex: 1;
  padding: 6px 12px;
  border: 1px solid var(--gray-800);
}

.prefix,
input {
  border-radius: 2px 4px 4px 0px;
}

input.hasPrefix {
  border-radius: 0px 4px 4px 0px !important;
}

.prefix .prefix-span {
  border-right: 0;
  border-radius: 4px 0px 0px 4px;
}

.input-group + .input-group {
  margin-top: 10px;
}
