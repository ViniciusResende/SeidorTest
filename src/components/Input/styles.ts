import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  border-radius: 2px 4px 4px 0px;
`;

export const PrefixSpan = styled.span`
  display: flex;
  align-items: center;
  padding: 6px 12px;

  font-size: 14px;
  color: var(--gray-800);
  background-color: var(--ice);
  box-shadow: 0px 2px 2px rgba(187, 204, 221, 0.4);
  border-right: 0;
  border-radius: 4px 0px 0px 4px;
`;

export const InputElement = styled.input`
  flex: 1;
  padding: 6px 12px;
  border: 1px solid var(--gray-800);
  border-radius: 2px 4px 4px 0px;

  &.hasPrefix {
    border-radius: 0px 4px 4px 0px !important;
  }
`;