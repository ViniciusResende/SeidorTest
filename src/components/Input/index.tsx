import React, { InputHTMLAttributes, useCallback } from 'react';

import { cpf, currency } from './masks';

import { Container, PrefixSpan, InputElement } from './styles';

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  mask: "cpf" | "currency";
  prefix?: string;
}

const Input: React.FC<InputProps> = ({ mask, prefix,...props}) => {
  
  const handleKeyUp = useCallback((e: React.FormEvent<HTMLInputElement>) => {
    if(mask === 'cpf'){
      cpf(e);
    }

    if(mask === "currency"){
      currency(e);
    }

  }, [mask])

  return(
    <Container>
      {prefix && <PrefixSpan>{prefix}</PrefixSpan>}
      <InputElement {...props} onKeyUp={handleKeyUp} className={prefix && "hasPrefix"}/>
    </Container>
  );
}

export default Input;