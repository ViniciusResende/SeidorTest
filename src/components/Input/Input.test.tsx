import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Input from './index';

test('Should render the custom input', () => {
  const tree = renderer
    .create(<Input mask="currency" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render a prefix div with span in case of having it', () => {
  const { getByText } = render(<Input mask="currency" prefix="R$"/>);
  const spanPrefix = getByText("R$");
  expect(spanPrefix).toBeInstanceOf(HTMLSpanElement);
})