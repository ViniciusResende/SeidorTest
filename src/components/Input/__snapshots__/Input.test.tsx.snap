// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the custom input 1`] = `
<div
  className="sc-bdfBwQ gMQfCg"
>
  <input
    className="sc-dlfnbm iBTgHz"
    onKeyUp={[Function]}
  />
</div>
`;
